spec/v2.spec.ts

```diff
@@ -348,6 +359,182 @@ describe('v2', () => {
+          const cloudFn = database.onValueDeleted(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const cloudEvent = cloudFnWrap().cloudEvent;
+          expect(cloudEvent).deep.equal({
+            // id and time are generated at runtime
+            id: cloudEvent.id,
+            time: cloudEvent.time,
+            specversion: '1.0',
+
+            data: cloudEvent.data,
+            instance: 'instance-1',
+            firebaseDatabaseHost: 'firebaseDatabaseHost',
+            ref: 'foo/baz',
+            location: 'us-central1',
+            params: {},
+            source: '',
+            type: 'google.firebase.database.ref.v1.deleted',
+          });
+        });
+
+        it('should use overridden data', () => {
+          const referenceOptions = {
+            ref: 'foo/bar',
+            instance: 'instance-1',
+          };
+          const cloudFn = database.onValueDeleted(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const dataVal = { snapshot: 'override' };
+          const data = makeDataSnapshot(dataVal, referenceOptions.ref);
+          const cloudEvent = cloudFnWrap({ data }).cloudEvent;
+
+          expect(cloudEvent.data.val()).deep.equal(dataVal);
+        });
+      });
+
+      describe('database.onValueUpdated()', () => {
+        it('should update CloudEvent appropriately', () => {
+          const referenceOptions = {
+            ref: 'foo/bar',
+            instance: 'instance-1',
+          };
+          const cloudFn = database.onValueUpdated(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const cloudEvent = cloudFnWrap().cloudEvent;
+          expect(cloudEvent).deep.equal({
+            // id and time are generated at runtime
+            id: cloudEvent.id,
+            time: cloudEvent.time,
+            specversion: '1.0',
+
+            data: cloudEvent.data,
+            instance: 'instance-1',
+            firebaseDatabaseHost: 'firebaseDatabaseHost',
+            ref: 'foo/bar',
+            location: 'us-central1',
+            params: {},
+            source: '',
+            type: 'google.firebase.database.ref.v1.updated',
+          });
+        });
+
+        it('should use overridden data', () => {
+          const referenceOptions = {
+            ref: 'foo/bar',
+            instance: 'instance-1',
+          };
+          const cloudFn = database.onValueUpdated(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const afterDataVal = { snapshot: 'after' };
+          const after = makeDataSnapshot(afterDataVal, referenceOptions.ref);
+
+          const beforeDataVal = { snapshot: 'before' };
+          const before = makeDataSnapshot(beforeDataVal, referenceOptions.ref);
+
+          const data = { before, after };
+          const cloudEvent = cloudFnWrap({ data }).cloudEvent;
+
+          expect(cloudEvent.data).deep.equal({ before, after });
```

Comment:
for consistency with the other tests, you could add
```
expect(cloudEvent.data.before.val()).deep.equal(beforeDataVal);
expect(cloudEvent.data.after.val()).deep.equal(afterDataVal);
```

feel free to ignore

Reply:
I agree with this, updated!